consensus/src/consensus_observer/observer.rs

```diff
@@ -166,28 +168,51 @@ impl ConsensusObserver {
         debug!(LogSchema::new(LogEntry::ConsensusObserver)
             .message("Checking consensus observer progress!"));
 
-        // Get the peer ID of the currently active subscription
+        // Get the peer ID of the currently active subscription (if any)
         let active_subscription_peer = self
             .active_observer_subscription
             .as_ref()
             .map(|subscription| subscription.get_peer_network_id());
 
         // If we have an active subscription, verify that the subscription
-        // is still healthy. If not, the subscription will be terminated.
-        if active_subscription_peer.is_some() {
-            self.check_active_subscription().await;
+        // is still healthy. If not, the subscription should be terminated.
+        if let Some(active_subscription_peer) = active_subscription_peer {
+            if let Err(error) = self.check_active_subscription().await {
```

Comment:
nit: it doesn't look like it's async function?

Reply:
Aah, nice, thanks!